File: gestion-tareas-frontend/src/app/models/tarea.ts

//Clase Tarea
export class Tarea {

//Propiedades de la tarea
    id?:number;
    titulo?:string;
    description?:string;
    completado?:boolean;

    //Constructor de la clase
    constructor(id: number = 0, titulo: string = '', description: string = ''){
        this.id=id;
        this.titulo=titulo;
        this.description=description;
        this.completado=false;
    }
}

File: gestion-tareas-backend/src/main/java/com/gestiontareas/entities/Tarea.java

package com.gestiontareas.entities;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;



//@Entity indica que esta clase es una entidad JPA y se mapeará a una tabla en la bd
@Entity
@AllArgsConstructor
@NoArgsConstructor
@Data

public class Tarea {
    
     //@Id marca este atributo como la clave primaria de la entidad
    @Id
     //@GeneratedValue especifica que el valor de esta clave primaria se generará automáticamente
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    // Atributo para almacenar el título de la tarea
    private String titulo;
    
    // Atributo para almacenar la descripción de la tarea
    private String description;
    
    // Atributo para indicar si la tarea está completada o no
    private boolean completado;
    
    
}